tests/bulkwritecommand/bulkwritecommand-updateOne-001.phpt

```diff
@@ -0,0 +1,110 @@
+--TEST--
+MongoDB\Driver\BulkWriteCommand::updateOne() should always encode __pclass for Persistable objects
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php skip_if_not_live(); ?>
+<?php skip_if_server_version('<', '8.0'); ?>
+<?php skip_if_not_clean(); ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+class MyClass implements MongoDB\BSON\Persistable
+{
+    private $id;
+    private $child;
+
+    public function __construct($id, ?MyClass $child = null)
+    {
+        $this->id = $id;
+        $this->child = $child;
+    }
+
+    public function bsonSerialize(): array
+    {
+        return [
+            '_id' => $this->id,
+            'child' => $this->child,
+        ];
+    }
+
+    public function bsonUnserialize(array $data): void
+    {
+        $this->id = $data['_id'];
+        $this->child = $data['child'];
+    }
+}
+
+$manager = create_test_manager();
+
+$document = new MyClass('foo', new MyClass('bar', new MyClass('baz')));
+
+$bulk = new MongoDB\Driver\BulkWriteCommand();
+// @TODO: Fatal error: Uncaught MongoDB\Driver\Exception\InvalidArgumentException: Invalid key '__pclass': update only works with $ operators and pipelines
+$bulk->updateOne(NS,
+    ['_id' => 'foo'],
+    $document,
+    ['upsert' => true]
+);
```

Comment:
We should wrap of documents in a `$set`, the simplest would be to do it in PHPLIB, but for the best compatibility we can do it in PHPC.

Reply:
Ok, I'll update the test to put the document in the `$set` operator.